File: c/src/coptic_months_data.c
#include "kiahk.h"
#include <string.h>

/*
 * Hand-maintained mirror of core/coptic_months.json.
 * Keep order identical (months 1..13) for cross-port test parity.
 */
const kiahk_coptic_month_record KIAHK_COPTIC_MONTHS[] = {
    { 1,  { "Thout",    "توت" } },
    { 2,  { "Paopi",    "بابة" } },
    { 3,  { "Hathor",   "هاتور" } },
    { 4,  { "Koiak",    "كيهك" } },
    { 5,  { "Tobi",     "طوبة" } },
    { 6,  { "Meshir",   "أمشير" } },
    { 7,  { "Paremhat", "برمهات" } },
    { 8,  { "Parmouti", "برمودة" } },
    { 9,  { "Pashons",  "بشنس" } },
    { 10, { "Paoni",    "بؤونة" } },
    { 11, { "Epip",     "أبيب" } },
    { 12, { "Mesori",   "مسرى" } },
    { 13, { "Nasie",    "نسيء" } },
};

const size_t KIAHK_COPTIC_MONTHS_COUNT =
    sizeof(KIAHK_COPTIC_MONTHS) / sizeof(KIAHK_COPTIC_MONTHS[0]);

kiahk_error kiahk_coptic_month_name(int month, const char *locale, const char **out) {
    if (out == NULL || locale == NULL) {
        return KIAHK_ERR_UNSUPPORTED_LOCALE;
    }
    if (month < 1 || month > 13) {
        return KIAHK_ERR_INVALID_COPTIC_MONTH;
    }
    const kiahk_coptic_month_record *rec = &KIAHK_COPTIC_MONTHS[month - 1];
    if (strcmp(locale, "en") == 0) {
        *out = rec->names.en;
        return KIAHK_OK;
    }
    if (strcmp(locale, "ar") == 0) {
        *out = rec->names.ar;
        return KIAHK_OK;
    }
    return KIAHK_ERR_UNSUPPORTED_LOCALE;
}

File: js/src/coptic-months-data.ts
/**
 * Hand-maintained mirror of core/coptic_months.json. Keep order identical
 * (months 1..13) for cross-port test parity. Inlined (not read from disk)
 * so this module works in both Node and browser environments.
 */
export interface CopticMonthRecord {
  month: number
  names: Record<string, string>
}

export const COPTIC_MONTHS: CopticMonthRecord[] = [
  { month: 1,  names: { en: 'Thout',    ar: 'توت' } },
  { month: 2,  names: { en: 'Paopi',    ar: 'بابة' } },
  { month: 3,  names: { en: 'Hathor',   ar: 'هاتور' } },
  { month: 4,  names: { en: 'Koiak',    ar: 'كيهك' } },
  { month: 5,  names: { en: 'Tobi',     ar: 'طوبة' } },
  { month: 6,  names: { en: 'Meshir',   ar: 'أمشير' } },
  { month: 7,  names: { en: 'Paremhat', ar: 'برمهات' } },
  { month: 8,  names: { en: 'Parmouti', ar: 'برمودة' } },
  { month: 9,  names: { en: 'Pashons',  ar: 'بشنس' } },
  { month: 10, names: { en: 'Paoni',    ar: 'بؤونة' } },
  { month: 11, names: { en: 'Epip',     ar: 'أبيب' } },
  { month: 12, names: { en: 'Mesori',   ar: 'مسرى' } },
  { month: 13, names: { en: 'Nasie',    ar: 'نسيء' } }
]
